docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+![ViewCommandSequenceDiagram.png](images\DeveloperGuide\ViewCommandSequenceDiagram.png)
+
+### Design considerations
+- The limit for the timespans are deliberately set to cover common timespans
+    - One can view up to 31 days / 4 weeks, as they each make up a month
+    - Similarly, one can view up to 12 months, as they make up a year
+    - In line with the average user of the target audience, the limit for 10 years was set as a soft limit
+- The setting of hiding the value of transactions that are ignored is deliberate, as it is the most prominent and
+  direct way for users to see this.
+    - In line with the purpose of the Ignore Command, -all flag will also not show this. Instead, users can view the
+      value via the export command, as elaborated below
+
+### Setting your monthly Budget Goal `setbudget`
+
+- The command `setbudget` is used to set the user's monthly budget goal
+    - Format: `setbudget VALUE` 
+    - Users can also turn off the feature at any time by setting `VALUE` to 0
+- Once a goal is present, user's will be reminded of how close they are to sticking to their budget, or how
+  much they have exceeded it by.
+- This can be seen at start-up and when the user makes any changes to their expenses for the month.
+
+![SetBudgetCommandSequenceDiagram.png](images\DeveloperGuide\SetBudgetCommandSequenceDiagram.png)
+
+### Editing an entry `edit`
+
+- When a command is given to edit a statement, the command is first parsed to check whether it follows the format of an
+  edit command: `edit INDEX ADDCOMMAND` or `edit INDEX FLAG NEWFIELD` or `edit INDEX FLAG` with the use of regex.
+- If the statement contains "add", then a check will be done on the remaining fields to see if it is a valid add
+  command.
+- Otherwise, methods specific to the flags will be used to validate the remaining fields using regex pattern
+- Commands in the correct format will then be used to create a FilterCommand object.
+- `RainyDay` will then call `execute` method in `Command`, where the transaction's specific field be edited or deleted
+  then added
+  into the financial report.
+
+![EditCommand.png](images\DeveloperGuide\EditCommand.png)
+
+### Filtering your data `filter`
+
+- When a command is given to filter a report by certain conditions, the command is first parsed to check whether it
+  follows the format of a filter command with the use of regex pattern
+    - Valid patterns are `filter -in`, `filter -out` , `filter -c <category>`,
+      `filter -d <description>` and `filter -date <DD/MM/YYYY>`
+- Commands in the correct format will then be parsed to create a FilterCommand object.
+- RainyDay will then call Command.execute(), where every entry in the financial report with the matching
+  conditions will be printed.
+- Information will be presented in a table format to help improve clarity for users.
+
+The sequence diagram for the implementation of filter is as shown below:
+
+![FilterCommand.png](images\DeveloperGuide\FilterCommand.png)
```

Comment:
This sequence was very confusing as it consisted of many if-else statements. Would the empty alt blocks mean the it does not perform any functions?